### DISCUSSION TITLE
I found a way to avoid recompiling when restarting the kernel.

### DISCUSSION THREAD

**Discussion by baiguoname** [evcxr/evcxr#336] [General]
Here is the process:
1. I install evcxr_jupyter kernel.
2. The `init.evcxr` has the following commands:
```
:sccache 1
:allow_static_linking 1
:linker mold
```
3. I remove all the commands from `init.evcxr` and restart the kernel
Then after that,  `:sccache` is still `true`(and `linker` is still `mold`), but each time I restart the kernel, there is no recompiling.
I thought this discovery is import for someone like me needing to restart the kernel frequently, it saves a lot of time.

**Comment by davidlattimore** [comment]
Thanks. `:allow_static_linking 1` is a new setting that restores the old behaviour. It's not yet released. I hadn't realised that using dynamic linking was causing sccache to not cache. After looking into it, it seems that sccache refuses to cache anything other than rlibs. Apparently this is because other kinds of libraries are linked using the system linker. I'm not sure why that prevents caching, but apparently it does.

**Comment by davidlattimore** [comment]
I added a caching mechanism directly within evcxr itself. You can now use `:cache 500` to enable a 500MB cache (on Linux this is in `~/.cache/evcxr`). Unlike using sccache, this cache works even when using dynamic linking (`:allow_static_linking 0`).
